src/helpers.js

```diff
@@ -12,6 +12,41 @@ import { processJSONLogicNode } from './jsonLogic';
 import { hasProperty } from './utils';
 import { buildCompleteYupSchema, buildYupSchema } from './yupSchema';
 
+/**
+ * List of custom JSF's attributes for field
+ * that are added dynamically after the first parsing.
+ */
+const dynamicInternalJsfAttrs = [
+  'isVisible', // Driven from conditionals state
+  'fields', // driven from group-array
+  'getComputedAttributes', // From json-logic
+  'computedAttributes', // From json-logic
+  'calculateConditionalProperties', // driven from conditionals
+  'calculateCustomValidationProperties', // To be deprecated in favor of json-logic
+  'scopedJsonSchema', // The respective JSON Schema
+];
+const dynamicInternalJsfAttrsObj = Object.fromEntries(
+  dynamicInternalJsfAttrs.map((k) => [k, true])
+);
+
+/**
+ *
+ * @param {Object} field - Current field attributes
+ * @param {Object} conditionalAttrs - Attributes from the matched conditional
+ * @param {Object} rootAttrs - Original field attributes from the root.
+ */
+function removeConditionalStaleAttributes(field, conditionalAttrs, rootAttrs) {
+  Object.keys(field).forEach((key) => {
+    if (
+      conditionalAttrs[key] === undefined &&
+      rootAttrs[key] === undefined && // Don't remove attrs that were declared in the root field.
```

Comment:
Yes, also the `dynamicInternalJsfAttrsObj` (which is the opposite of your initial solution).

Reply:
> [@brennj] Just one small thing needs to be changed and its only a lint error. Otherwise, tests are fantastic and I'm so happy we have this now.

Cool! I'm gonna wait for the internal pipelines to be green before merging this one.